runtime/storage-proxy.js

```diff
@@ -352,6 +377,17 @@ class VariableProxy extends StorageProxyBase {
     if (this._barrier != null) {
       if (update.barrier == this._barrier) {
         this._barrier = null;
+
+        // HOLY LAYERING VIOLATION BATMAN
```

Comment:
Similar to above comment -- is this long term or will it go away at some point, if so, what will drive that.